### ISSUE TITLE
Missing some dependencies in setup.py

### ISSUE THREAD

**Issue by SimonMaenaut** [mikekatz04/LISAanalysistools#18]
Hi, in order to incorporate this package as a dependency in a larger python project (where I'd like to use `lisatools.sensitivity`), pip needs to be aware of the minimal dependencies needed to use this package. Often these are included in `setup.py`.

At the moment these aren't in `setup.py` so `pip install lisaanalysistools` in a clean env. doesn't include dependencies.

<details>
<summary>List of missing dependencies based in install.sh and a quick scan of the python files in lisatools</summary>

```
setup_requires=[
"cython",
"numpy",
],
install_requires=[
"cython",
"numpy",
"h5py",
"scipy",
"matplotlib",
"fastemriwaveforms",
"fastlisaresponse",
"requests",
"tqdm",
]
```
</details>

To help, I would be more than willing to set up a pull request with changes.

Also I'm curious to learn why `python>=3.12` is specifically required.

**Comment by mikekatz04** [comment]
>=3.12 is required because of the static typing docstrings, etc. It was intended as temporary. Going to work on this so it is not the case. 

Will update dependencies.

**Comment by mikekatz04** [comment]
Dependencies packages have been added. Will be included in version >=1.0.6
